generators/chipyard/src/main/scala/TestHarness.scala

```diff
@@ -39,7 +42,9 @@ class TestHarness(implicit val p: Parameters) extends Module with HasHarnessSign
   // dutReset assignment can be overridden via a harnessFunction, but by default it is just reset
   val dutReset = WireDefault(if (p(GlobalResetSchemeKey).pinIsAsync) reset.asAsyncReset else reset)
 
-  ldut.harnessFunctions.foreach(_(this))
-
+  lazyDut match { case d: HasTestHarnessFunctions =>
+    d.harnessFunctions.foreach(_(this))
```

Comment:
What are these used for now? Aren't they all applied below?

Reply:
The clock/reset of ChipTop is still connected by a `harnessFunction`. I left `harnessFunctions` around for unusual cases which can't be captured by a `HarnessBinder`.